### DISCUSSION TITLE
Write-Only Property or something else?

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3247] [Questions]
I think this is the second time we hit a point in one of our existing projects where our parent object needs to run a rule based on a property in a child object.

We don't need (to persist) that property in the parent object other than to run a validation rule in the parent object.  Perhaps our initial object design was wrong : ).

Is there a way we could accomplish this other than a 'dummy' write-only property in the parent object?
Kind Regards

**Comment by lwmorris067** [comment]
Something like having a Person class with an Address collection... a rule to make sure you don't have duplicate addresses? Or something like Invoice with line items. Invoice having a Total property that must be under a certain amount?

**Reply by Chicagoan2016** [reply]
Something like a Person object with Address object (single child) and the parent object 'Person' needs to run a business rule based off of ZipCode which is a property of Address object.
(of course a simplified version of our 'use case')

**Reply by rockfordlhotka** [reply]
Can you use a per-type rule instead of a per-property rule? In other words, don't attach the rule to a property in the parent class; instead attach it to the parent class itself.

There's no easy way to show the user the results of per-type rules in some UI frameworks (Windows Forms specifically), but in other UI frameworks it isn't terribly difficult. And you can use the option to check object rules when manually calling `CheckRules`.

Is that sort of what you are after?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Thank you, should I  create a per-type rule and then use 'Target' inside  the rule to find the property on the child business object?

**Reply by rockfordlhotka** [reply]
That's my thought. The real trick is that you want to trigger the rule to run based on a `ChildChanged` event right? And you don't want a property in the parent class, so you need a rule that you can trigger, so that leads me to think a per-type rule.

The rule itself would need to use the `Target` property to access the object graph so it can access the child and/or grandchild objects in the graph.
